streampipes-extensions/streampipes-processors-filters-jvm/src/main/java/org/apache/streampipes/processors/filters/jvm/processor/duration/ConditionalTimeFilterProcessor.java

```diff
@@ -0,0 +1,291 @@
+                .build()
+            )
+            .requiredSingleValueSelection(
+                Labels.withId(DURATION_UNIT_ID),
+                Options.from(
+                    "Seconds",
+                    "Minutes",
+                    "Hours"
+                )
+            )
+            .requiredIntegerParameter(
+                Labels.withId(DURATION_ID)
+            )
+            .requiredSlideToggle(Labels.withId(CONDITION_VALUE_ID), true)
+            .requiredSingleValueSelection(Labels.withId(TRIGGER_MODE_ID),
+                Options.from(
+                    TRIGGER_MODE_ON_EVENT,
+                    TRIGGER_MODE_ON_TIMER
+                )
+            )
+            .requiredSingleValueSelection(
+                Labels.withId(OUTPUT_MODE_ID),
+                Options.from(
+                    "Fire Once",
+                    "Fire Continuously"
+                )
+            )
+            .outputStrategy(OutputStrategies.append(
+                EpProperties.timestampProperty(ORIGINAL_TIMESTAMP_FIELD),
+                EpProperties.timestampProperty(PROCESSING_TIMESTAMP_FIELD),
+                EpProperties.longEp(Labels.withId(TIME_DIFFERENCE_FIELD), TIME_DIFFERENCE_FIELD, SO.NUMBER)
+            ))
+            .build()
+    );
+  }
+
+  @Override
+  public void onPipelineStarted(IDataProcessorParameters params,
+                                SpOutputCollector collector,
+                                EventProcessorRuntimeContext runtimeContext) {
+    IDataProcessorParameterExtractor extractor = params.extractor();
+    this.filterFieldSelector = extractor.mappingPropertyValue(FILTER_FIELD_ID);
+    this.timestampFieldSelector = extractor.mappingPropertyValue(TIMESTAMP_FIELD_ID);
+    this.duration = extractor.singleValueParameter(DURATION_ID, Integer.class);
+    this.timeUnit = extractor.selectedSingleValue(DURATION_UNIT_ID, String.class);
+    this.outputMode = extractor.selectedSingleValue(OUTPUT_MODE_ID, String.class);
+    this.triggerMode = extractor.selectedSingleValue(TRIGGER_MODE_ID, String.class);
+    this.conditionToMeet = extractor.slideToggleValue(CONDITION_VALUE_ID);
+
+    if (this.triggerMode.equals(TRIGGER_MODE_ON_TIMER)
+        && (this.executorService == null || this.executorService.isShutdown())) {
+      this.executorService = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    this.resetState();
+  }
+
+  @Override
+  public void onEvent(Event event, SpOutputCollector out) {
+    try {
+      boolean currentCondition = event.getFieldBySelector(this.filterFieldSelector).getAsPrimitive().getAsBoolean();
+
+      if (currentCondition == this.conditionToMeet) {
+        if (triggerMode.equals(TRIGGER_MODE_ON_TIMER)) {
+          handleOnTimer(event, out);
+        } else {
+          handleOnEventArrival(event, out);
+        }
+      } else {
+        resetState();
+      }
+    } catch (SpRuntimeException e) {
+      // Handle exceptions, e.g., by logging
```

Comment:
You can use the StreamPipes logger within the `onPipelineStarted` method by accessing it through the `runtimeContext`:

```java
runtimeContext.getLogger().error(e);
```

These logs will later be visible in the UI, enabling better observability. This is the functionality I referred to during our discussion on monitoring.